### PR TITLE
[Core] Embeds toggle for channels.

### LINKED ISSUE #3152: [Core] Channel-specific setting for `embed_requested` feature
# Feature request

#### Select the type of feature you are requesting:

- [ ] Cog
- [ ] Command
- [x] API functionality

#### Describe your requested feature

e.g. `[p]embedset channel` to toggle channels for channel. We already take channel object to `bot.embed_requested` so should be relatively easy if I get ok on it

### PR DESCRIPTION
**Pull request by Stonedestroyer**
### Type

- [ ] Bugfix
- [X] Enhancement
- [X] New feature

### Description of the changes
Resolves #3152. 

Channels have priority and then guilds.

### CODE REVIEW

**Review by Jackenmen — DISMISSED** [review]
LGTM, however marking this as blocked,
CC: @mikeshardmind leaving this for you specifically to merge as I heard from you that you might want to postpone this until after we will cache this stuff and I'm not entirely sure if this is still the case.

Aaaand, Travis check bugged, it's duplicated for some reason so I can't even merge this if I want apparently...

**Comment by Stonedestroyer** [comment]
@jack1142 I forgot the decorators for guild only and server admin. I have now added that and the travis build should now also work.

**Review by Jackenmen — APPROVED** [review]
Still needs Sinbad's sign-off, but the PR is okay™

**Comment by mikeshardmind** [comment]
I'd like to look more at the performance impact on this to see if we can use it now and add the caching later as planned, or if it really should just wait. Will evaluate that tonight.

**Review by mikeshardmind — APPROVED** [review]

**Pull request merged**
